File: src/main/resources/templates/javascript/solution/src/bubblesort.js
export default class BubbleSort {
    /**
     * Sorts dates with BubbleSort.
     *
     * @param input {Date[]} the array of Dates to be sorted
     */
    performSort(input) {
        for (let i = input.length - 1; i >= 0; i--) {
            for (let j = 0; j < i; j++) {
                if (input[j].valueOf() > input[j + 1].valueOf()) {
                    const temp = input[j];
                    input[j] = input[j + 1];
                    input[j + 1] = temp;
                }
            }
        }
    }
}

File: src/main/resources/templates/ruby/solution/src/bubble_sort.rb
# frozen_string_literal: true

class BubbleSort
  # Sorts dates with BubbleSort.
  # @param input [Array<Date>] the Array of Dates to be sorted
  def perform_sort(input)
    (input.length - 1).downto(0) do |i|
      (0...i).each do |j|
        input[j], input[j + 1] = input[j + 1], input[j] if input[j] > input[j + 1]
      end
    end
  end
end
